packages/proxy/lib/http/util/prerequests.ts

```diff
@@ -123,16 +129,39 @@ export class PreRequests {
 
     if (pendingRequest) {
       debugVerbose('Incoming pre-request %s matches pending request. %o', key, browserPreRequest)
-      clearTimeout(pendingRequest.timeout)
-      pendingRequest.callback(browserPreRequest)
+      if (!pendingRequest.timedOut) {
+        clearTimeout(pendingRequest.timeout)
+        pendingRequest.callback({
+          ...browserPreRequest,
+          cdpRequestWillBeSentTimestamp: browserPreRequest.cdpRequestWillBeSentTimestamp,
+          cdpRequestWillBeSentReceivedTimestamp: browserPreRequest.cdpRequestWillBeSentReceivedTimestamp,
+          proxyRequestReceivedTimestamp: pendingRequest.proxyRequestReceivedTimestamp,
+          cdpLagDuration: browserPreRequest.cdpRequestWillBeSentReceivedTimestamp - browserPreRequest.cdpRequestWillBeSentTimestamp,
+          proxyRequestCorrelationDuration: Math.max(browserPreRequest.cdpRequestWillBeSentReceivedTimestamp - pendingRequest.proxyRequestReceivedTimestamp, 0),
```

Comment:
Suggest pulling these calculations out of the `if` statement so they can just be reused below.

Reply:
Done: [`9581c1f` (#27976)](https://github.com/cypress-io/cypress/pull/27976/commits/9581c1f795772798725207a36bfe803b09da02c2)